frontend/src/home/components/Homepage.jsx

```diff
@@ -25,26 +25,42 @@ export default class Homepage extends Component {
     }
 
     static propTypes = {
-        dispatch: PropTypes.func.isRequired,
         onChangeLocation: PropTypes.func.isRequired,
         showOnboarding: PropTypes.bool.isRequired,
-        user: PropTypes.object.isRequired
+        user: PropTypes.object.isRequired,
+        activity: PropTypes.array,
+        hasSampleDataset: PropTypes.bool,
+        fetchActivity: PropTypes.func.isRequired,
+        fetchRecentViews: PropTypes.func.isRequired,
+        fetchHasSampleDataset: PropTypes.func.isRequired
     };
 
+    async componentDidMount() {
+        try {
+            await this.props.fetchHasSampleDataset();
```

Comment:
this feels a bit icky to me.  we are now requiring an api call to get the list of databases every time someone hits the homepage.  it would be nice if we could think of another way to get at this information.